src/ScottPlot5/ScottPlot/Colormaps/IColormap.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ScottPlot.Colormaps
+{
+    public interface IColormap
+    {
+        string Name { get; }
+        Color GetColor(double intensity, Range? domain);
```

Comment:
Same comment as above. Also I think the XML docs for GetColor should be pretty clear about how it works and what the corner cases are (e.g., transparency)

I admit things are flying around and I haven't set-up a meaningful/rigid test project yet (what's there is really thrown together), but the special NaN behavior would be a good candidate for the Beyonce rule: Regarding special features (and protecting from them getting removed by accident), _if you like it then you shoulda put a test on it_ 😜